lib/iris/tests/unit/coords/test_Coord.py

```diff
@@ -324,6 +324,12 @@ def test_numeric_nd(self):
         with self.assertRaises(ValueError):
             coord.collapsed()
 
+    def test_collapsed_overflow(self):
+        coord = DimCoord(points=np.array([1493892000, 1493895600, 1493899200],
```

Comment:
I suggest using `np.iinfo(np.int32).max` to get appropriate numbers here.  Up to you, just think it makes it clearer.